File: server/medicore.apps/src/model/articleCategoryModel.ts
import { IEntityBase } from './IEntityBase';

export class ArticleCategoryModel implements IEntityBase {
    id: number;
    parentId: number;
    categoryName: string;
    slug: string;
    isVisible: boolean;
    articleCount: number;

    constructor() {
        this.id = 0;
        this.parentId = 0;
        this.categoryName = null;
        this.slug = null;
        this.isVisible = false;
        this.articleCount = 0;
    }
}

File: server/dna.core/dna.core.service/Models/ArticleCategoryModel.cs
using dna.core.service.Models.Abstract;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace dna.core.service.Models
{
    public class ArticleCategoryModel : IModelBase
    {
        public int Id { get; set; }
        public int ParentId { get; set; }

        [Required, MaxLength(255)]
        public string Name { get; set; }

        [Required, MaxLength(255)]
        public string Slug { get; set; }

        public bool IsVisible { get; set; }

        // The image url path
        [MaxLength(100)]
        public string Image { get; set; }

        public int ArticleCount { get; set; }
    }
}
